agent/connect/ca/provider_aws_test.go

```diff
@@ -84,7 +91,7 @@ func testSignAndValidate(t *testing.T, p Provider, rootPEM string, intermediateP
 
 	err = connect.ValidateLeaf(rootPEM, leafPEM, intermediatePEMs)
 	require.NoError(t, err)
-	requireTrailingNewline(t, leafPEM)
+	//requireTrailingNewline(t, leafPEM)
```

Comment:
I don't think this is right, it pass the tests but the same issue that we had with vault, is happening with AWS provider. The cert don't have a trailing new line and when we create a chain out of it, it will be not valid. I think we need to apply the same fix (using `EnsureTrailingNewline`) to the AWS provider

Reply:
hey @dhiaayachi thanks for having a look here.

> we need to apply the same fix (using EnsureTrailingNewline) to the AWS provider

totally! I tried calling that out in my comment and tbh, that's why I branched off a PR to add the root cert ttl option for the aws provider from the "main" pr (#11428)

I can focus on adding `EnsureTrailingNewline` as needed then add the root cert ttl option.

Reply:
This has been addressed as part of https://github.com/hashicorp/consul/pull/11470